### DISCUSSION TITLE
@entangle appears to break dom-diffing

### DISCUSSION THREAD

**Discussion by ju5t** [livewire/livewire#7095] [7. Bugs (archived)]
### Livewire version

v3.0.5

### Which PHP version are you using?

PHP 8.2

### Repository URL

https://github.com/ju5t/livewire-items-reappearing

### Steps To Reproduce

We have a weird dom-diffing issue that we found while upgrading to v3.

It seems that whenever we we delete multiple rows in a table, they reappear and it's impossible to remove them. After some testing I found that this is related to custom blade components. We have several, most of them are using AutoNumeric. Edit: I have done some more testing but it seems to be affecting blade components with `@entangle`. For example:

```blade
<div x-data="{ value: @entangle($attributes->wire('model')) }">
    <input
        type="text"
        {{ $attributes->whereDoesntStartWith('wire:model') }}
    />
</div>
```

Will keep items in the dom, while this works:

```blade
<div x-data>
    <input
        type="text"
        {{ $attributes->whereDoesntStartWith('wire:model') }}
    />
</div>
```

I've included a test repository. It's not using any models, so migrations shouldn't be necessary.

1. Clone the repository.
2. Go to the welcome page in your browser.
3. Hit add a couple of times. 5 rows should be enough.
4. Try to delete all the rows by clicking the `delete`-button.

This will delete most, but will leave some of them there. We did not have this issue in v2. Maybe we were just lucky? If I'm doing something wrong please let me know.

<s>I didn't know how to show this in a test, sorry for that.</s>

I just found https://github.com/livewire/livewire/blob/c9e99b2016707d137fbc966c46eaa60ac048870c/src/Features/SupportEntangle/BrowserTest.php#L128

This is slightly different from our issue as it's not using a Blade-component. I will try and have a go at it tomorrow to see if I can send in a PR with a failing test.

### Additional information

The commit which introduces the issue: https://github.com/ju5t/livewire-items-reappearing/commit/bf6db54c9f534c94a6a8f3b55d9b3512c75efa8a

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by satoved** [comment]
See https://github.com/livewire/livewire/pull/6833

**Reply by ju5t** [reply]
I don't know how I have not seen that. Thanks for the link and I have closed my PR in favour of yours.
